File: frontend/src/application/commands/products/createProduct/CreateProductCommand.ts
import ICommand from "../../ICommand";
import ICreateProductResult from "./ICreateProductResult";

export default class CreateProductCommand implements ICommand<ICreateProductResult> {
    __returnType: ICreateProductResult = null!;

    constructor(props: {
        name: string,
        price: number,
        description: string,
        images: string[]
    }) {
        this.name = props.name
        this.price = props.price
        this.description = props.description
        this.images = props.images
    }

    public name: string;
    public price: number;
    public description: string;
    public images: string[];
}

File: Application/Handlers/Products/Create/CreateProductCommand.cs
using Application.ErrorHandling.Application;
using MediatR;
using OneOf;

namespace Application.Handlers.Products.Create;

public class CreateProductCommand : IRequest<OneOf<CreateProductResult, List<PlainApplicationError>>>
{
    public CreateProductCommand(string name, decimal price, string description, List<string> images)
    {
        Name = name;
        Price = price;
        Description = description;
        Images = images;
    }

    public string Name { get; set; }
    public decimal Price { get; set; }
    public string Description { get; set; }
    public List<string> Images { get; set; }
}